addons/hr_holidays/hr_holidays.py

```diff
@@ -321,19 +321,32 @@ def onchange_date_to(self, cr, uid, ids, date_to, date_from):
 
         return result
 
+    def add_follower(self, cr, uid, ids, employee_id, context=None):
+        employee = self.pool['hr.employee'].browse(cr, uid, employee_id, context=context)
+        if employee and employee.user_id:
+            self.message_subscribe_users(cr, uid, ids, user_ids=[employee.user_id.id], context=context)
```

Comment:
the method `message_subscribe_users` will browse again the user_id to find the id of the partner, As you already have a recordset, you can fetch the id of the partner and call `message_subscribe` directly